apis/i18n.d.ts

```diff
@@ -0,0 +1,46 @@
+import { csn, entity } from './cds'
+
+
+declare class I18nBundle {
+  constructor(options: I18nFilesOptions)
+  for(key: number | string | object, locale?: string | object, args?: object): string | undefined
+  at(key: number | string | object, locale?: string | object, args?: object): string | undefined
+  file: string
+  files: I18nFiles
+  get defaults(): Record<string, string>
+  get fallback(): Record<string, string>
+  key4(definition: entity): string
+  texts4 (locale: string): Texts
+  translations4 (...locales : 'all' | string[]) : { [locale]: Texts }
```

Comment:
Is this supposed to mean:

1. "you can pass either a single string `'all'`, or a bunch of arbitrary strings" or
2. "you can pass arbitrary strings, one of which is `'all'`

?

In case 1., we would need to work with overloads:

```ts
  translations4 (locales: 'all'): …
  translations4 (...locales : string[]) : …
```

as `'all'` is not a valid rest parameter, and in case of 2., it should look like this:

```ts
  translations4 (...locales : ('all' | string)[]) : …
```
(although the literal string `'all'` will get swallowed by the more general `string`)

I am also not sure about the return type. This looks like it was taken [verbatim from CAPire](https://cap.cloud.sap/docs/node.js/cds-i18n#translations4-locales), but does not pass as valid TS, to my knowledge (can't check out the code right now). Did you mean:

```ts
  translations4 (…) : { [locale: string]: Texts }
```
?

Reply:
It seems that case 1 is right. I've added the overload.